### DISCUSSION TITLE
original: error: foreign key constraint "accounts_files_fkey" cannot be implemented

### DISCUSSION THREAD

**Discussion by FatNerdPeng** [sequelize/sequelize#14624] [Help & Questions] [answered]
hello,I want to create the following data model:
![image](https://user-images.githubusercontent.com/33616855/173277964-4256526f-7f4c-4741-960a-61692505623d.png)
So I wrote:
![image](https://user-images.githubusercontent.com/33616855/173278018-cd19a8a1-898a-41b4-8f46-2fdf3ee6f4aa.png)
![image](https://user-images.githubusercontent.com/33616855/173278056-2d9d9aa9-fa8c-465b-87fd-c6dfabe2c4bc.png)
![image](https://user-images.githubusercontent.com/33616855/173278089-81d6baeb-da82-48e0-b76d-d79c7c962399.png)
and then:
![image](https://user-images.githubusercontent.com/33616855/173278129-ebe97867-544e-4567-9fa3-c936b3e75e36.png)
![image](https://user-images.githubusercontent.com/33616855/173278167-f43b94d4-4a09-4e49-8f00-4245d766589f.png)
![image](https://user-images.githubusercontent.com/33616855/173278188-528d8a07-0893-491e-aa36-5606723d3c6a.png)
why?
my sequelize version:6.20.1

**Comment by ephys** [comment, marked as answer]
`sync` is an async operation, you need to await it. Consider moving it to somewhere after all of your models have been loaded and/or use sequelize.sync instead

**Reply by FatNerdPeng** [reply]
thanks,but I made the same mistake as you said,and then I saw this:
![image](https://user-images.githubusercontent.com/33616855/173293544-908dfa3b-c803-416e-8012-6df5674de3c1.png)
I wonder if it's because table account's attributes files is array of string,but table file's attributes cid is string?

**Reply by ephys** [reply]
You can't have an array of foreign keys. If you need a many-to-many you need a through table

**Reply by FatNerdPeng** [reply]
Thank you. According to your tips, I successfully solved the problem with one-to-many
